doc/bgp_error_handling/BGP_Route_Error_Handling_Arlo.md

```diff
@@ -0,0 +1,241 @@
+       T - Table, v - VNC, V - VNC-Direct, A - Babel, D - SHARP,
+       F - PBR,
+       > - selected route, * - FIB route, # - Not installed in hardware
+
+K>* 0.0.0.0/0 [0/0] via 10.59.128.1, eth0, 09:44:37
+C>* 4.1.1.0/24 is directly connected, Ethernet4, 00:01:48
+C>* 10.1.0.1/32 is directly connected, lo, 09:44:37
+C>* 10.59.128.0/20 is directly connected, eth0, 09:44:37
+B># 21.21.21.21/32 [20/0] via 4.1.1.2, Ethernet4, 00:00:07
+```
+A new command has been introduced for seeing the failed routes as follows  
+  show {ip | ipv6} route not-installed [prefix/mask]
+```
+sonic# show ip route not-installed
+Codes: K - kernel route, C - connected, S - static, R - RIP,
+       O - OSPF, I - IS-IS, B - BGP, E - EIGRP, N - NHRP,
+       T - Table, v - VNC, V - VNC-Direct, A - Babel, D - SHARP,
+       F - PBR,
+       > - selected route, * - FIB route # - not installed in hardware
+B> # 22.1.1.1/32 [20/0] via 4.1.1.2, Ethernet4, 00:00:20
+B> # 22.1.1.2/32 [20/0] via 4.1.1.2, Ethernet4, 00:00:20
+B> # 30.1.1.1/32 [20/0] via 4.1.1.2, Ethernet4, 00:00:20
+B> # 30.1.1.2/32 [20/0] via 4.1.1.2, Ethernet4, 00:00:20
+B> # 30.1.1.3/32 [20/0] via 4.1.1.2, Ethernet4, 00:00:20
+B> # 30.1.1.4/32 [20/0] via 4.1.1.2, Ethernet4, 00:00:20
+B> # 30.1.1.5/32 [20/0] via 4.1.1.2, Ethernet4, 00:00:20
+B> # 30.1.1.6/32 [20/0] via 4.1.1.2, Ethernet4, 00:00:20
+B> # 30.1.1.7/32 [20/0] via 4.1.1.2, Ethernet4, 00:00:20
+B> # 30.1.1.8/32 [20/0] via 4.1.1.2, Ethernet4, 00:00:20
+```
+### 3.7.4 Debug Commands
+In order to retry the installation of failed routes from Zebra, a clear command has been provided.  
+  clear {ip | ipv6} route {not-installed | <prefix/mask>}
+  ```
+sonic# clear ip route 
+  not-installed  not installed in hardware
+sonic# clear ip route not-installed 
+  <cr>        
+  A.B.C.D/M   ipv4 prefix with mask
+  X:X::X:X/M  ipv6 prefix with mask
+```
+The above command will send route add message for the failed route from Zebra to fpmsyncd. 
+
+### 3.7.5 REST API Support
+
+# 4 Flow Diagrams
+ ![BGP](images/bgp_error_handling_flow1.png "Figure 1: High level module interaction for route install error notification")
+
+__Figure 1: High level module interaction for route install error notification__ 
+
+![BGP](images/bgp_error_handling_flow2.png "Figure 2: Module flow for route add error notification")
+
+__Figure 2: Module flow for route add error notification__
+
+![BGP](images/bgp_error_handling_flow3.png "Figure 3: Module flow for route install success notification")
+
+__Figure 3: Module flow for route add success notification__
+ ![BGP](images/bgp_error_handling_flow4.png "Figure 4: Module flow for route delete success/fail notification")
+
+__Figure 4: Module flow for route delete success/fail notification__
+ 
+# 5 Serviceability and Debug
+
+
+# 6 Warm Boot Support
+
+# 7 Scalability
+
+# 8 Unit Test
+
+The UT testcases are as follows:
+
+|**Test-Case ID**|**Description**|**Status**|**Comments**|  

```

Comment:
What is the scope of the QuickTests? Is it covered only happy paths alone? do you have numbers with non-happy path scenarios?

Reply:
Yes, it covers the happy paths alone. I am still testing the feature using quicktest and will cover the non-happy paths also. From the BGP perspective, for routes that have failed to be installed in hardware, the route will not be sent to its peers and traffic will not be received at the receivers.